packages/alchemy/src/client/rpcClient.ts

```diff
@@ -1,14 +1,40 @@
 import { createBundlerClient, type ConnectionConfig } from "@alchemy/aa-core";
-import { http, type Chain } from "viem";
+import { http, type Chain, type HttpTransportConfig } from "viem";
 import { AlchemyChainSchema } from "../schema.js";
+import { VERSION } from "../version.js";
 import type { ClientWithAlchemyMethods } from "./types.js";
 
+export type DynamicFetchOptions = HttpTransportConfig["fetchOptions"] & {
+  get headers(): HeadersInit | undefined;
+  set headers(newFetchOptions: HeadersInit | undefined);
+};
+
+const createDynamicFetchOptions = (
+  initialFetchOptions: HttpTransportConfig["fetchOptions"] = {}
+): DynamicFetchOptions => {
+  let headers = {
+    ...initialFetchOptions.headers,
+  };
+
+  return {
+    ...initialFetchOptions,
+    get headers() {
+      return headers;
+    },
+    set headers(newHeaders) {
+      headers = newHeaders;
+    },
+  };
+};
```

Comment:
Similarly, why is this getter and setter different than the default behavior of a field? This function looks like it might be equivalent to
```ts
const createDynamicFetchOptions = (
  initialFetchOptions: HttpTransportConfig["fetchOptions"] = {}
): DynamicFetchOptions => {
  return {
    ...initialFetchOptions,
    headers: { ...initialFetchOptions.headers },
  };
};
```

Reply:
yea I think you might be right. will try it without the getter setters. initially I thought I might need a mutable variable that the getter and setter accessed instead of the default behaviour. I'll give it a shot now that I have tests and see what happens

Reply:
ok tested it out and you're right this works just fine by mutating the object directly. updated everything to make it cleaner